### DISCUSSION TITLE
Filtering away documents based on metadata property

### DISCUSSION THREAD

**Discussion by mic2zar** [ravendb/ravendb#13507] [Q&A] [answered]
We are experimenting with different features of RavenDB from the RavenWorkshop repository. There is this proposal on how to handle soft deletion:
https://github.com/DejanMilicic/RavenWorkshop/blob/main/Northwind/Northwind/Features/Metadata/Metadata.cs

What we could not find is how to filter away in auto query such a soft deleted documents. Theoretically this solution on “Querying metadata using Lucene in C#” from 2018 should address our request:
https://blog.hildenco.com/2018/07/querying-ravendb-metadata.html
, but it is somehow broken. Here is where we are stuck:
                return session
                    .Advanced.AsyncDocumentQuery<T>()
                    .Where(!@"metadata.IsDeleted: true")
                    .ToListAsync();
or 
                return session
                    .Query<T>()
                    .Where(!@"metadata.IsDeleted: true")
                    .ToListAsync();

In our case writing own index is rather not an option. Such a filtering in our system would be used in the context of Visitor pattern (every object is adding own properties to the final index). I cannot guess upfront of all properties from inherited objects that I would need to include, so in the long run having handwritten index would be not maintainable and probably only auto index would fit into to address our problem.

**Comment by ayende** [comment, marked as answer]
You are using syntax that is meant for the 3.5 version.

What you want is probably:

```
return session
.Advanced.AsyncDocumentQuery()
.WhereNotEqua("@metadata.IsDeleted", true)
.ToListAsync();
```

That said, if soft delete is part of your domain, you might be better off doing that as a domain property. 
If you are using static indexes, you can also filter them directly in the index map.
